File: Scripts/newTask/call-an-object-method/c++/call-an-object-method.cpp


// Static
//MyClass::method(someParameter);

// Instance
//myInstance.method(someParameter);


#include <iostream>

using namespace std;

class someMethod {
		int setNumber;
	public:	
		void setValue(int );
};

void someMethod::setValue (int value) {
	setNumber = value;
}


int main () {

	someMethod  objectCall;

	for (int i = 0; i < 1000000000; ++i)
		objectCall.setValue(100);

	return 0;
}

File: Scripts/newTask/call-an-object-method/c-sharp/call-an-object-method.cs
// Static
//MyClass.Method(someParameter);
 
// Instance
//myInstance.Method(someParameter);

using System;

class MyClass{

  // instance variable
  private int variable;  // Note: instance variables are usually "private"

  /**
  * The constructor
  */
  public MyClass(){
    // creates a new instance
  }

  /**
  * A method
  */
  public void someMethod(int value){
   this.variable = value;
  }

public static void Main(String[] args){

	MyClass obj = new MyClass();
	for (int i = 0; i < 1000000000; ++i) {
		obj.someMethod(1);
	}
}

}
